cookbook/console/style.rst

```diff
@@ -0,0 +1,371 @@
+.. index::
+   single: Console; Style commands
+
+How to Style a Console Command
+==============================
+
+One of the most boring tasks when creating console commands is to deal with the
+styling of the command's input and output. Displaying titles and tables or asking
+questions to the user involves a lot of repetitive code.
+
+Consider for example the code used to display the title of the following command::
+
+    // src/AppBundle/Command/GreetCommand.php
+    namespace AppBundle\Command;
+
+    use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+    use Symfony\Component\Console\Input\InputInterface;
+    use Symfony\Component\Console\Output\OutputInterface;
+
+    class GreetCommand extends ContainerAwareCommand
+    {
+        // ...
+
+        protected function execute(InputInterface $input, OutputInterface $output)
+        {
+            $output->writeln('<info>Lorem Ipsum Dolor Sit Amet</>');
+            $output->writeln('<info>==========================</>');
+            $output->writeln('');
+
+            // ...
+        }
+    }
+
+Displaying a simple title requires three lines of code, to change the font color,
+underline the contents and leave an additional blank line after the title. Dealing
+with styles is required for well-designed commands, but it complicates their code
+unnecessarily.
+
+In order to reduce that boilerplate code, Symfony commands can optionally use the
+**Symfony Style Guide**. These styles are implemented as a set of helper methods
+which allow to create *semantic* commands and forget about their styling.
+
+Basic Usage
+-----------
+
+In your command, instantiate the :class:`Symfony\\Component\\Console\\Style\\SymfonyStyle`
+class and pass the ``$input`` and ``$output`` variables as its arguments. Then,
+you can start using any of its helpers, such as ``title()``, which displays the
+title of the command::
+
+    // src/AppBundle/Command/GreetCommand.php
+    namespace AppBundle\Command;
+
+    use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+    use Symfony\Component\Console\Helper\SymfonyStyle;
+    use Symfony\Component\Console\Input\InputInterface;
+    use Symfony\Component\Console\Output\OutputInterface;
+
+    class GreetCommand extends ContainerAwareCommand
+    {
+        // ...
+
+        protected function execute(InputInterface $input, OutputInterface $output)
+        {
+            $io = new SymfonyStyle($input, $output);
+            $io->title('Lorem Ipsum Dolor Sit Amet');
+
+            // ...
+        }
+    }
+
+.. tip::
+
+    You can use any name for the variable that stores the Symfony styles, but
+    it's recommended to name it ``$io`` because it's concise and conveys the
+    notion that it interacts both with the input and the output of the command.
```

Comment:
I'm -1 on explicitely recommending the variable name. 90% of the users will already use `$io` just because they copy/past the code examples and the other 10% probably had a good reason (or they just wanted something else) to give it another name.